Create vagrant homestead box

mreeves1 opened the pull request:
- Make it easier to get tattle running

@g76r - I should probably complete the TODO in after.sh but wanted to get some feedback first

mreeves1 committed:
Create vagrant homestead box

- Make it easier to get tattle running

mreeves1 committed:
Create vagrant homestead box

- Some documentation and cleanup

g76r commented:
Hi and thank you for your contribution !

I need time to review it because I'm still new to Vagrant but I will look at it in depth.

Concerning filling up the root .gitignore with all that stuff, I think it should be better to keep a small .gitignore containing only patterns directly related to tattle. You may use .git/info/exclude or even use git config --global core.excludesfile (especially for the apple .DS_double and the like).

mreeves1 commented:
Sounds good. I think I should still keep the .vagrant and vendor exclusions as those are just dependencies but I will test some more and just have the bare minimum. Let me know if you have any questions!

g76r commented:
As I said I'm new to Vagrant but as far as I understand it should be possible to put the Vagrant and Homestead files in a subdirectory to avoid commiting all these files in the repo root.

Also, I'm not sure why you need to commit a composer.json file (maybe Homestead needs it), but I don't think composer.lock couldn't be recreated at deployment time (hence commiting it seems odd to me).

Apart from that, I didn't get it working, and I think that you should write (short) instructions, not just write "All you really need to do is to follow homestead instructions" in the Readme.md. I don't know if the full instructions should take place in the main Readme.md (probably, since I expect them to be at most 10 lines of commands) or in another readme in Vagrant-Homestead subdirectory. But anyway, either I really try to use Vagrant the wrong way, or there are a few "sudo apt-get" and "sudo vagrant add" that should be done before running tattle.

Maybe it should start with that:

```
sudo aptitude install vagrant virtualbox # only If you're not already using vagrant and virtualbox
git clone -b make_vagrant_box https://github.com/mreeves1/Tattle.git # of course in real readme it will be on master branch and repos
cd Tattle
vagrant box add laravel/homestead
```

But if I try this, I've got an error:

```
There was an error loading a Vagrantfile. The file being loaded
and the error message are shown below. This is usually caused by
a syntax error.

Path: /home/gb/test/Tattle/Vagrantfile
Message: cannot load such file -- /home/gb/test/Tattle/vendor/laravel/homestead/scripts/homestead.rb
```

Therefore something is missing I guess.

mreeves1 commented:
Thanks for your feedback; I'll try to take a look this weekend. You're running this on linux? Which distribution/version so I can try to reproduce?

g76r commented:
Debian 8

Commit:
Create vagrant homestead box

- Fix error re: server address
- Improve instructions
- Remove needless git ignore lines

mreeves1 commented:
Sorry for the delay, I've been busy. I improved the instructions and got this to work on mac and windows. I will test linux soon but there were definitely some small bugs and poor instructions. Your specific issue was due to composer not having been run yet. I'll test some more using only my instructions to see if I missed anything else.
